### PR TITLE
Use preact instead of react

### LINKED ISSUE #633: Investigate if we can use preact instead of react
**Describe exactly what you would like to see in an upcoming release**

`preact` is much smaller than React. We use both Vue and React. The latter is because of GraphiQL, which is a React application.

It's more challenging to use preact + graphiql as in our case the GraphiQL is using React, and we need — somehow — to tell GraphiQL that preact is react.

**Additional context**

At the moment we have:

```
react-dom
117.2kB

react
6.1kB
```

I tried installing `preact` and `preact-compat`, aliasing in Webpack, then running the UI, but unfortunately GraphiQL appears to have troubles using the React API with preact.

```
⠋  Building for development...

 WARNING  Compiled with 18 warnings                                                                                                         9:52:34 AM

 warning  in ./node_modules/@apollo/client/react/hooks/useSubscription.js

"export 'useContext' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useMutation.js

"export 'useContext' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/utils/useBaseQuery.js

"export 'useContext' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useReactiveVar.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useReactiveVar.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useSubscription.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useSubscription.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useMutation.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/utils/useBaseQuery.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/utils/useBaseQuery.js

"export 'useEffect' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/utils/useBaseQuery.js

"export 'useReducer' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useSubscription.js

"export 'useRef' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useMutation.js

"export 'useRef' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/utils/useBaseQuery.js

"export 'useRef' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/utils/useDeepMemo.js

"export 'useRef' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useReactiveVar.js

"export 'useState' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useSubscription.js

"export 'useState' was not found in 'react'

 warning  in ./node_modules/@apollo/client/react/hooks/useMutation.js

"export 'useState' was not found in 'react'

  File                        Size                                                       Gzipped

  dist/js/chunk-vendors.js    19193.63 KiB                                               4360.94 KiB
  dist/js/4.js                3377.90 KiB                                                813.47 KiB
  dist/js/1.js                3340.13 KiB                                                788.07 KiB
  dist/js/5.js                2552.53 KiB                                                467.34 KiB
  dist/js/app.js              1754.58 KiB                                                306.69 KiB
  dist/js/2.js                1747.10 KiB                                                324.45 KiB
  dist/js/6.js                351.45 KiB                                                 58.57 KiB
  dist/js/8.js                144.94 KiB                                                 28.24 KiB
  dist/js/7.js                137.09 KiB                                                 28.06 KiB
  dist/js/0.js                109.44 KiB                                                 34.87 KiB
  dist/js/3.js                97.25 KiB                                                  17.94 KiB
  dist/js/10.js               68.45 KiB                                                  15.38 KiB
  dist/js/11.js               48.34 KiB                                                  10.39 KiB
  dist/js/9.js                42.36 KiB                                                  9.94 KiB
  dist/fonts.css              3.64 KiB                                                   0.77 KiB
```

**Pull requests welcome!**

### PR DESCRIPTION
**Pull request by kinow**
These changes close #633 

It works! Before:

```
react-dom
117.2kB

react
6.1kB
```

Screenshot from 2021-03-29 10-26-40

After the change:

```
preact
19.5kB
```

Screenshot from 2021-03-29 10-22-53

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Does not need tests (why? for GraphiQL only, confirm it's working accessing GraphiQL after re-building the project).
- [x] No change log entry required (why? e.g. invisible to users).
- [x] No documentation update required.

### CODE REVIEW

File: vue.config.js [hunk 1]
```diff
@@ -46,6 +46,14 @@ module.exports = {
...
   },
   chainWebpack: config => {
+    config
+      .resolve
+      .alias
+      .set('react', 'preact/compat')
+    config
+      .resolve
+      .alias
+      .set('react-dom', 'preact/compat')
```
Comment (kinow) [inline vue.config.js:56]: This way Webpack will provide the link for the applications using React, and they will get preact instead.

File: package.json [hunk 2]
```diff
@@ -65,8 +65,8 @@
     "babel-plugin-istanbul": "^6.0.0",
     "bufferutil": "^4.0.3",
     "chai": "^4.3.4",
-    "cross-env": "^7.0.3",
     "circular-dependency-plugin": "^5.2.2",
+    "cross-env": "^7.0.3",
```
Comment (kinow) [inline package.json:69]: Yarn sorted alphabetically, no change in `cross-env` actually :+1:

**Comment by kinow** [comment]
Low priority, for after the beta release.

**Review by hjoliver — APPROVED** [review]
Nice, and for GraphiQl only so not really much potential for incompatibility in future?

**Comment by kinow** [comment]
> Nice, and for GraphiQl only so not really much potential for incompatibility in future?

Not in the near future I think. `preact` is not new, and has a good and growing community - https://github.com/preactjs/preact

But in case that happens, we just replace by `react` and `react-dom`, so easy to revert if necessary.

**Review by oliver-sanders — APPROVED** [review]

**Pull request merged**